### PR TITLE
[4.0] Use the app response so we get headers sent on error

### PR DESCRIPTION
**Pull request by wilsonge**
Send the app response in the exception handler through `$app->respond` as per @mbabker 's suggestion. This means headers are correctly returned for example. This is more than just a subtlety and required to make webservices work (see https://github.com/joomla-projects/gsoc18_webservices/pull/63)

## Testing Instructions
Test that 404 etc pages work

### CODE REVIEW

File: libraries/src/Exception/ExceptionHandler.php [hunk 1]
```diff
@@ -136,16 +139,17 @@ public static function render(\Throwable $error)
 
 			if ($isCli)
 			{
-				echo $data;
+				/** @var CliApplication $app */
+				$app->out($data);
```
Comment (mbabker) [inline libraries/src/Exception/ExceptionHandler.php]: `Joomla\Console\Application` doesn't have this `out` method, so this is only going to work for apps extending from `Joomla\CMS\Application\CliApplication`.  You'll need another if/else here.

**Review by mbabker — CHANGES_REQUESTED** [review]

**Comment by ghost** [comment]
Added "[4.0]" in Title.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/20431">issues.joomla.org/tracker/joomla-cms/20431</a>.</sub>

File: libraries/src/Exception/ExceptionHandler.php (hunk 1 above)
Comment (wilsonge) [reply libraries/src/Exception/ExceptionHandler.php]: Ok reverted back ye olde way for now

**Review by mbabker — APPROVED** [review]

**Comment by carlitorweb** [comment]
I have tested this item :white_check_mark: successfully on 2273abab5e590a68bbf5da58a2c7be261a79dc77<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/20431">issues.joomla.org/tracker/joomla-cms/20431</a>.</sub>

**Comment by laoneo** [comment]
I have tested this item :white_check_mark: successfully on 2273abab5e590a68bbf5da58a2c7be261a79dc77<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/20431">issues.joomla.org/tracker/joomla-cms/20431</a>.</sub>

**Comment by laoneo** [comment]
Thanks!

**Pull request merged**